sw/device/silicon_creator/manuf/lib/sram_program.ld

```diff
@@ -0,0 +1,98 @@
+/* Copyright lowRISC contributors. */
+/* Licensed under the Apache License, Version 2.0, see LICENSE for details. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/**
+ * Linker script for an OpenTitan SRAM program.
+ *
+ * Portions of this file are Ibex-specific.
+ */
+
+OUTPUT_ARCH(riscv);
+
+/**
+ * Indicate that there are no dynamic libraries, whatsoever.
+ */
+__DYNAMIC = 0;
+
+INCLUDE hw/top_earlgrey/sw/autogen/top_earlgrey_memory.ld
+
+_stack_end = ORIGIN(ram_main) + LENGTH(ram_main);
+
+/**
+ * DV Log offset.
+ *
+ * Note: This definition is required by the `sw/device/info_sections.ld` script.
+ */
+_dv_log_offset = 0x10000;
+
+/* We do not use this information but it used to prevent the linker from discarding
+ * the crt */
+ENTRY(sram_start);
+
+SECTIONS {
+  /**
+   * SRAM programs consist only of a `.data` section that contains all input
+   * `.rodata`, `.data`, `.text`, and `.bss` sections.
+   */
```

Comment:
What's the rationale of having everything in data section?

Reply:
This PR does not rewrite the linker script, this is just a copy of the existing one with BSS split in its own section.
The SRAM programs are loaded as one binary blob and the memory protection unit is not used to differentiate between code and data. The upcoming PR will change the situation a little bit in that the ELF file will finally be used to load them. But at the moment creating different sections for code and data brings no benefits.

Reply:
I think part of the issue was that SRAM program were loaded with hardcoded constants for pretty much everything (so one could not adjust the global pointer for each program). Now that SRAM programs have a CRT, this requirement goes away. This could be an opportunity to change it here if you want to suggest a better arragement?
Maybe something like:
```
    KEEP(*(.srodata))
    KEEP(*(.srodata.*))
    KEEP(*(.sdata))
    KEEP(*(.sdata.*))
    // define global pointer based on the small data here?
    KEEP(*(.rodata))
    KEEP(*(.rodata.*))
    KEEP(*(.data))
    KEEP(*(.data.*))
```

Reply:
another option is to address this in a follow up, as I'm guessing this impacts other linker scripts in the repo too? (Note, we will not be changing the ROM linker script in the immediate future however, as we are about tag a release candidate for the discrete shuttle tapeout.)

Reply:
I think this would make sense. Since we cannot touch the ROM, this makes most linker scripts off-limits. It probably makes more sense to change all of them at once? @nbdd0121 do you want to create an issue about this and explain what the problem is and how we could improve the situation?

Reply:
ROM linker script is fine, since we don't have a mutable data section anyway.